Library/Homebrew/livecheck/strategy.rb

```diff
@@ -119,25 +119,48 @@ def self.page_headers(url)
 
       # Fetches the content at the URL and returns a hash containing the
       # content and, if there are any redirections, the final URL.
+      # If `curl` encounters an error, the hash will contain a `:messages`
+      # array with the error message instead.
       #
       # @param url [String] the URL of the content to check
+      # @param user_agent [Symbol, String] the symbol for a corresponding user
+      #   agent or a user agent string
       # @return [Hash]
-      sig { params(url: String).returns(T::Hash[Symbol, T.untyped]) }
-      def self.page_content(url)
+      sig { params(url: String, user_agent: T.any(Symbol, String)).returns(T::Hash[Symbol, T.untyped]) }
+      def self.page_content(url, user_agent: :default)
         original_url = url
 
-        # Manually handling `URI#open` redirections allows us to detect the
-        # resolved URL while also supporting HTTPS to HTTP redirections (which
-        # are normally forbidden by `OpenURI`).
-        begin
-          content = URI.parse(url).open(redirect: false, &:read)
-        rescue OpenURI::HTTPRedirect => e
-          url = e.uri.to_s
-          retry
+        args = curl_args(
+          "--compressed",            # Request compressed transfer encoding
+          "--include",               # Include HTTP response headers in output
+          "--location",              # Follow redirects
+          "--connect-timeout", "10", # Max time allowed for connection (secs)
+          "--max-time", "15"         # Max time allowed for transfer (secs)
```

Comment:
This says "what" these are but not "why". The latter would be more useful.

Reply:
These are simply the explanations of the options from the `curl` usage information, as I felt that the intention with each was straightforward. From what I've seen, no other areas in brew have explanatory comments when `curl` options are specified, so this felt like a reasonable compromise between no comments and the verbosity required to explain the intention behind the options.

I generally agree on the issue of "why" vs. "what" but I don't think there's enough room to explain this without placing the comments on separate lines, which could make the code harder to read. I don't think anyone who's engaging with this code will have trouble understanding why these options are used but this is what I imagine it would take to explain the "why":

```ruby
args = curl_args(
  # Compressed transfer encoding reduces transfer size
  "--compressed",
  # Include HTTP response headers in output, so we can parse location
  # headers to identify the final URL after any redirections
  "--include",
  # Follow redirections to accommodate mirrors and allow URLs that have
  # moved to continue to resolve
  "--location",
  # cURL's default timeout can be up to two minutes, so we need to
  # set our own timeout settings to avoid a lengthy wait
  "--connect-timeout", "10", # Max time allowed for connection (secs)
  "--max-time", "15"         # Max time allowed for transfer (secs)
)
```

If you prefer this (or suggest an alternative) I can add it in but it doesn't really feel necessary to me compared to the brief comments here.

Reply:
Please either:
- remove these comments:
- make them explain "why" and remove e.g. "Max time allowed for connection" and other "what" comments.

Reply:
As mentioned in [my previous reply on this topic](https://github.com/Homebrew/brew/pull/10834#discussion_r594649316), adequately explaining the "why" ends up being more verbose and can make the code harder to read. Borrowing from my previous example, we may end up with something like this:

```ruby
args = curl_args(
  "--compressed",
  # Include HTTP response headers in output, so we can identify the
  # final URL after any redirections
  "--include",
  # Follow redirections to handle mirrors, relocations, etc.
  "--location",
  # cURL's default timeout can be up to two minutes, so we need to
  # set our own timeout settings to avoid a lengthy wait
  "--connect-timeout", "10",
  "--max-time", "15"
)
```

Is this closer to what you're looking for in the latter approach? If not, I'll just remove the comments entirely for the sake of simplicity.

Reply:
> As mentioned in [my previous reply on this topic](https://github.com/Homebrew/brew/pull/10834#discussion_r594649316), adequately explaining the "why" ends up being more verbose and can make the code harder to read.

I read your previous reply. I don't agree that comments explaining "why" make code harder to read and, if they do, they are worth just removing. "what" comments never add value.



> Is this closer to what you're looking for in the latter approach?

Yes, this works, thanks.